### PR TITLE
NSDate testing

### PR DESCRIPTION
**Pull request by konDeichmann**
##### Untested functions in `NSDate+DKHelper`

```Objective-c
- (void)logCurrentDateWithDateStyleAndAllTimeStyle:(NSDateFormatterStyle)dateStyle;
- (void)logAllFormats;

+ (NSString * _Nonnull)gregorianCalendarIdentifier DEPRECATED_MSG_ATTRIBUTE("use NSCalendarIdentifierGregorian");
```
Because they are deprecated or just not testable (`void` function)

##### 1 Bugfix was made:
`currentDayDate ` was setting the current settings (for GMT+1 the seconds were set to 1h), but seconds should be always 0.

### CODE REVIEW

**Comment by kevindelord** [comment]
Please also update the readme

File: Example/DKHelperTests/NSDateTests.swift [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+	func testShouldReturnValidDateByAddingMinute() {
+		if let date = today?.dateByAddingMinuteInterval(1) {
+			XCTAssertEqual(today?.laterDate(date), date)
+		} else {
+			XCTFail()
+		}
+	}
+
+	func testShouldReturnValidDateByAddingSecond() {
+		if let date = today?.dateByAddingSecondInterval(1) {
```
Comment (kevindelord) [inline Example/DKHelperTests/NSDateTests.swift]: shouldn't you check that the difference is really one second ?

File: Example/DKHelperTests/NSDateTests.swift [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+	func testShouldReturnValidDateByAddingHour() {
+		if let date = today?.dateByAddingHourInterval(1) {
+			XCTAssertEqual(today?.laterDate(date), date)
+		} else {
+			XCTFail()
+		}
+	}
+
+	func testShouldReturnValidDateByAddingMinute() {
+		if let date = today?.dateByAddingMinuteInterval(1) {
```
Comment (kevindelord) [inline Example/DKHelperTests/NSDateTests.swift]: could you maybe also test with `0`, `60`and maybe negative numbers?

File: Example/DKHelperTests/NSDateTests.swift [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+		}
+	}
+}
+
+// MARK: - ISO Format
+
+extension NSDateTests {
+
+	func testShouldReturnValidAndCorrectISOStringFromDate() {
+		let date = NSDate(fromString: "04.12.2015 - 14:27:11", format: "dd.MM.yyyy - HH:mm:ss")
```
Comment (kevindelord) [inline Example/DKHelperTests/NSDateTests.swift]: could you also tests with wrong format?

File: Example/DKHelperTests/NSDateTests.swift [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+		let dayName = date?.dayName()
+		let monthName = date?.monthName()
+		XCTAssertNotNil(date)
+		XCTAssertNotNil(dayName)
+		XCTAssertNotNil(monthName)
+		XCTAssertEqual(dayName, "Freitag", "expect german localization")
+		XCTAssertEqual(monthName, "Dezember", "expect german localization")
+	}
+
+	func testShouldReturnValidAndCorrectDateDisplayStrings() {
```
Comment (kevindelord) [inline Example/DKHelperTests/NSDateTests.swift]: please rename the function `test_Should...` to be consistent with the other files.

File: Example/DKHelperTests/NSDateTests.swift (hunk 2 above)
Comment (konDeichmann) [inline Example/DKHelperTests/NSDateTests.swift]: done

File: Example/DKHelperTests/NSDateTests.swift (hunk 3 above)
Comment (konDeichmann) [inline Example/DKHelperTests/NSDateTests.swift]: done

File: Example/DKHelperTests/NSDateTests.swift (hunk 4 above)
Comment (konDeichmann) [inline Example/DKHelperTests/NSDateTests.swift]: done
